Extend the technical pillar intro + add summary of software engineering (#290)
the-teachingRSE-project/competencies: competencies.md

The role of an RSE lies somewhere on the spectrum between that of a researcher
(the "R") and a software engineer (the "SE") and, therefore, requires

> competencies in both fields. RSEs have a background both in research and

Reviewer: I don't think background is the right word here since most have their background in one of both. 
Maybe something along the lines of

"RSEs typically have a background in research or software engineering but have obtained broader knowledge in both fields"

\newcommand{\skillsection}[1]{\hypertarget{skills-#1}{%
\subsubsection{\glsentrydesc{#1} (\texorpdfstring{\glsentrytext{#1}}{#1})}\label{skills-#1}}}

Besides skilled researchers, research software engineers are also competent software engineers.
As such, they ideally can solve complex software engineering problems and design software as a user-oriented, future-proof product.
The technical skills required by an RSE overlap to a large extent with
 the common fundamental software engineering skills (see, e.g., @Landwehr2017),
 but put greater emphasis on aspects related to achieving good scientific practice and to serving special needs of research software.
These skills include requirements analysis, design, construction, testing, program analysis, and maintenance of software.

> On the other hand, RSEs also know how to make research software adhere to the \ac{FAIR} principles,

Reviewer 2: ```suggestion
On the other hand, RSEs also know how to make research software adhere to (FAIR4RS [@FAIR4RS]) principles,
```

> On the other hand, RSEs also know how to make research software adhere to the (FAIR4RS [@FAIR4RS]) principles,

Reviewer 2: ```suggestion
On the other hand, RSEs also know how to make research software adhere to the \ac{FAIR} principles [@FAIR4RS],
```

 but put greater emphasis on aspects related to achieving good scientific practice and to serving special needs of research software.
These skills include requirements analysis, design, construction, testing, program analysis, and maintenance of software.
On the other hand, RSEs also know how to make research software adhere to the \ac{FAIR} principles,
 and how to achieve different levels of research software reusability (see, e.g., @ChueHong2014),
 while they have deeper understanding of the scientific context around the research software projects they work on.
To reflect this, the technical skills listed below complement competencies regarding the standard life cycle of software development (as summarised in \autoref{subsec:technical-general}) with RSE-specific focus skills. 

\subsubsection{Classical software engineering skills}\label{subsec:technical-general}

To summarise the vast range of the skills a software engineer is typically equipped with,
we refer to the Guide to the Software Engineering Body of Knowledge (@swebok_2014).
Because they stand between fields, RSEs are often stronger in topics more commonly

> encountered in research software contexts (e.g., mathematical and engineering foundations)

Reviewer 3: Can we add something from outside STEM here, to be more inclusive?

Reviewer 4: the rather cool project of managing audience participation during a concert presented at the deRSE conf springs to mind.

Although I think here the point is that SWEBOK puts maths and algo design as optional skill for a 'normal' SE and one of the distinguishing factors is that an RSE is probably naturally more clued up about those.

Reviewer 2: https://en.wikipedia.org/wiki/Formal_language

Reviewer 4: I think we should keep this as those two items are contrasted with SE economics

than in other (e.g., software engineering economics), but they bring a solid level of competence
in all software engineering topics. Therefore, RSEs can set and analyse software requirements in the context of
open-ended, question-driven research. They can design software so that it can sustainably grow, often
in an environment of rapid turnover of contributors. They are competent in implementing
solutions themselves in a wide range of technologies fit for different scientific applications.
They can formulate and implement various types of tests, they can independently maintain software
and automate operations of the integration and release process. They can
provide working, scalable, and future-proof solutions in a professional context and with common
project and software management techniques, adapted to the needs of the research environment.

> Finally, as skilled scientists seasoned in computing applications, they have strong mathematical, computing, and engineering foundations,

Reviewer 3: I think this is too strong. There are a lot of RSEs that don't need that much of math - since their research does not involve much math in the first place. They wouldn't call it 'computing' either, as they might not compute that much, but for instance deal with large amounts of text or images.
In addition, if we leave in the last part of the sentence ("to the degree every SE needs to have"), then this essentially says: you better are on the same level as someone with an SE degree to be an RSE - excluding large parts of our community.
Thus, I do vote to drop this entire sentence ("Finally ... until the end")

> Finally, as skilled scientists seasoned in computing applications, they have strong mathematical, computing, and engineering foundations,

Reviewer 2: ```suggestion
Finally, as skilled scientists seasoned in computing applications, they have strong mathematical, computing, and engineering foundations,
```
Finally, as skilled researchers they combine the necessary foundations from their domain with software engineering for their applications.

> to the degree every software engineer needs to have to develop complex software projects.

Reviewer 2: ```suggestion
```

\newcommand{\skillsection}[1]{\hypertarget{skills-#1}{%
\subsubsection{\glsentrydesc{#1} (\texorpdfstring{\glsentrytext{#1}}{#1})}\label{skills-#1}}}

Besides skilled researchers, research software engineers are also competent software engineers.
As such, they solve complex software engineering problems, and design software as a user-oriented, future-proof product.
The technical skills required by an RSE overlap to a large extent with the common 
fundamental software engineering skills (see, e.g., @Landwehr2017), but put greater
emphasis on aspects related to achieving good scientific practice.

> RSEs are competent in requirements analysis, design, construction, testing, analysis, and testing of software.

Reviewer 4: ```suggestion
RSEs are competent in requirements analysis, design, construction, and testing of software.
```

Too much testing and analysis

Author: That was definitely a typo, thanks!

listed below complement competencies regarding the standard life cycle of
software development (as summarised in \autoref{subsec:technical-general}) with RSE-specific focus skills. 

\subsubsection{Classical software engineering skills}\label{subsec:technical-general}

To summarise the vast range of the skills a software engineer is typically equipped,
we refer to the Guide to the Software Engineering Body of Knowledge (@swebok_2014).
Because they stand between fields, RSEs are stronger in topics more commonly
encountered in research software contexts (e.g., mathematical and engineering foundations)
than in other (e.g., software engineering economics), but they bring a solid level of competence
in all software engineering topics. Therefore, RSEs can set and analyse software requirements in the context of
open-ended research. They can design software so that it can sustainably grow, often
in an environment of rapid turnover of contributors. They are competent in implementing
solutions themselves in a wide range of technologies fit for different scientific applications.
They can formulate and implement various types of tests, they can independently maintain software

> and automate operations of the integration and release process. They are engineers that can

Author: Let's not use the naked "engineering"/"engineers" term lightly, since in Germany, people assume certified Mechanical Engineer or similar.

fundamental software engineering skills (see, e.g., @Landwehr2017), but put greater
emphasis on aspects related to achieving good scientific practice and to serving special needs of research software.
RSEs are competent in requirements analysis, design, construction, testing, program analysis, and maintenance of software.
But they also know, for example, how to make research software adhere to the 
\ac{FAIR} principles, and how to achieve different levels of research
software reusability (see, e.g., @ChueHong2014), while they have deeper understanding of the scientific context
around the research software projects they work on. To reflect this, the technical skills
listed below complement competencies regarding the standard life cycle of
software development (as summarised in \autoref{subsec:technical-general}) with RSE-specific focus skills. 

\subsubsection{Classical software engineering skills}\label{subsec:technical-general}

To summarise the vast range of the skills a software engineer is typically equipped with,
we refer to the Guide to the Software Engineering Body of Knowledge (@swebok_2014).

> Because they stand between fields, RSEs are stronger in topics more commonly

Reviewer: maybe "are often stronger in topics more commonly..."?

Reviewer 4: the rather cool project of managing audience participation during a concert presented at the deRSE conf springs to mind.

Although I think here the point is that SWEBOK puts maths and algo design as optional skill for a 'normal' SE and one of the distinguishing factors is that an RSE is probably naturally more clued up about those.

encountered in research software contexts (e.g., mathematical and engineering foundations)
than in other (e.g., software engineering economics), but they bring a solid level of competence
in all software engineering topics. Therefore, RSEs can set and analyse software requirements in the context of
open-ended, question-driven research. They can design software so that it can sustainably grow, often
in an environment of rapid turnover of contributors. They are competent in implementing
solutions themselves in a wide range of technologies fit for different scientific applications.
They can formulate and implement various types of tests, they can independently maintain software
and automate operations of the integration and release process. They can
provide working, scalable, and future-proof solutions in a professional context and with common
project and software management techniques, adapted to the needs of the research environment.
Finally, as skilled researchers they combine the necessary foundations from their domain with software engineering skills to develop complex software.

There are many ways to categorise the competencies of an RSE. We chose to

> distribute these competencies over the three categories to reflect the fact

Reviewer 2: ```suggestion
distribute these competencies over the three pillars to reflect the fact
```

that RSEs are both competent researchers (the research skills, Section
@sec:research-skills) and software engineers (the software/technical skills,

> Section @sec:software-skills). The third categories (communication skills,

Reviewer 2: ```suggestion
Section @sec:software-skills). The third category (communication skills,
```
